core/src/main/java/org/apache/iceberg/SnapshotParser.java

```diff
@@ -89,8 +89,12 @@ static void toJson(Snapshot snapshot, JsonGenerator generator)
       generator.writeEndArray();
     }
 
-    // schema ID might be null for snapshots written by old writers
-    if (snapshot.schemaId() != null) {
+    // For test purposes, we want to emulate an old writer; we do this if the system property
+    // iceberg.snapshot.no-schema-id is set to true.
+    boolean noSchemaId = SystemProperties.getBoolean("iceberg.snapshot.no-schema-id", false);
```

Comment:
I'm open to suggestions for other ways of getting the metadata to be written without `schema-id` in the `snapshot`s, or even other ways of testing this. But this was a simple, unintrusive way (very little code change needed, in particular, no API changes needed).